### DISCUSSION TITLE
What's code?

### DISCUSSION THREAD

**Discussion by nichtich** [hedyorg/hedy#1590] [Archive]
I'm trying to improve the German translation and stumbled upon the term "code", first used in level 2:

>  In level 2 we learn something new. You can name a word with `is`. You can use name anywhere in your code and it will be replaced by Hedy, like this

The start page introduces "commands"

> Commands are instructions for a computer

It is assumed the student knows what a "program" is. But what's "code", especially compared to "command" and "program"? Should be wetter avoid this term or explain it?

**Comment by Felienne** [comment]
> I'm trying to improve the German translation and stumbled upon the term "code", first used in level 2:
> 
> > In level 2 we learn something new. You can name a word with `is`. You can use name anywhere in your code and it will be replaced by Hedy, like this

Thanks for thinking about how to make everything clear to kids! Looking at the texts we indeed use both "code" and "program". I think that is fine but you are right, maybe we should define that somewhere.

> The start page introduces "commands"
> 
> > Commands are instructions for a computer
> 
> It is assumed the student knows what a "program" is.

> But what's "code", especially compared to "command" and "program"? Should be wetter avoid this term or explain it?

Code I think is quite clear for English and also Dutch speakers, we also say "coding" and "code" means a secret language. So I would not avoid it, but in German the connotation might very well be different!

Also, I don't think we explicitly say this somewhere (we should!) but most of our users are schools, not individual kids. So a teacher would of course start with some explanation about what code is and why it is a good/fun thing to learn. That being said; an introductory sentence for both code and program will be helpful, I agree!

**Reply by nichtich** [reply]
How about something like

> Commands are instructions for a computer. These instructions make code and you can use them to write programs.
